packages/contracts-bedrock/contracts/L2/L1Block.sol

```diff
@@ -74,5 +83,49 @@ contract L1Block is Semver {
         batcherHash = _batcherHash;
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
+
+        _setL1BlockHash(_number, _hash);
+    }
+
+    /// @notice Returns the hash of the L1 block at the specified block number.
+    /// @dev Requires the block number to be within the range of stored block hashes.
+    /// @param _block The number of the block for which to return the hash.
+    /// @return The hash of the L1 block.
+    function getL1BlockHash(uint256 _block) external view returns (bytes32) {
+      uint64 _last_number = number;
+      uint64 _first_number;
+      bytes32 _hash;
+
+      assembly {
+        _first_number := add(sub(_last_number, BLOCK_HASHES_SIZE), 1)
+      }
+
+      if (_block < _first_number) {
+        revert L1BlockBlockHashIndexOutOfBound();
```

Comment:
This might underflow if we deploy to a network that has less than 50k blocks already. This is not the case for optimism mainnet or testnets, however it's worth to bring awareness. Are we fine with that?